scalar/scalar.go

```diff
@@ -228,6 +219,19 @@ func AppendToBuilder(bldr array.Builder, s Scalar) {
 			AppendToBuilder(sb.FieldBuilder(i), sc)
 		}
 
+	case arrow.MAP:
+		mb := bldr.(*array.MapBuilder)
+		mb.Append(true)
+
+		v := s.(*Map).Value
+		_ = v
```

Comment:
Maps are a bit tricky to append.
See this code: https://github.com/cloudquery/cloudquery/blob/main/plugins/destination/clickhouse/typeconv/ch/values/map_test.go